app.ts

```diff
@@ -0,0 +1,149 @@
+function InsertValue(value: string, id: string) {
+    const writable: string = '<td id=\"' + id + '\" align="center">' + value.toString() + '</td>';
+    $("#ContentBody").append(writable);
+}
+function UpdateColumnNames(value: string) {
+    const writable: string = '<th align="center">' + value.toString() + '</th>';
+    $("#HeaderRow").append(writable);
+}
+function placeRecords(fromID: number, toID: number) {
+    $("#ContentBody").empty();
+    getRecords(fromID, toID)
+        .then((records: Array<Array<string>>) => {
+            for (var rowIndex in records) {
+                $("#ContentBody").append('<tr id="' + records[rowIndex][0] + '">');
+                for (var columnIndex in records[rowIndex]) {
+                    InsertValue(records[rowIndex][columnIndex], records[rowIndex][0]);
+                }
+                $("#ContentBody").append('</tr>');
+            }
+        });
+    return [fromID as number, toID as number];
+}
+
+function placeRecordsFromCursor(cursor: Array<number>) {
+    cursor = cursor.sort((a,b) => {return a-b});
+    return placeRecords(cursor[0], cursor[1])
+}
+//#endregion
+
+//#region Handlers
+function getPageContent(fromID: number, toID: number) {
+    const cursor = [fromID as number, toID as number];
+    $("#HeaderRow").empty();
+    getColumnNames()
+        .then((columns: Array<string>) => {
+            for (var columnIndex in columns) {
+                UpdateColumnNames(columns[columnIndex]);
+            }
+        });
+    placeRecords(fromID, toID);
+    return cursor;
+}
+
+function toNumber(input: string | number) {
+    if (typeof input === "string") {
+        return parseInt(input);
+    }
+    else if (typeof input === "number") {
+        return input;
+    }
+    else {
+        return 0;
+    }
+}
+function nextPageResize(previousCursor: Array<number>): number {
+        $("form-content").height()
+        var fromID = toNumber(previousCursor.sort((a, b) => {return a - b})[0]);
+        var toID = toNumber(previousCursor.sort((a, b) => {return a - b})[1]);
+
+        for (var i = fromID; i <= toID; i++) {
+            var element = ($("#" + i.toString()).offset() as JQueryCoordinates).top as number;
+            var documentHeight = ($(window).innerHeight() as number) - ($("form-content").innerHeight() as number) - 10;
+
+            if ( element >= documentHeight ){
+                return i;
+            } else {
+                continue
+            };
+        }
+        return toID
+}
+
+function previousPageResize(previousCursor: Array<number>): Array<number> {
+    var fromId: number = previousCursor[0] - (nextPageResize(previousCursor) - previousCursor[0]);
+    var toId: number = fromId + step;
+    return [previousCursor[0] - (nextPageResize(previousCursor) - previousCursor[0]), toId]
+}
+//#endregion
+//#endregion
+
+var step: number = 37;
+var previousCursor: Array<number>;
+
+window.onload = () => {     
+    previousCursor = getPageContent(0,37)
```

Comment:
I think you should rather use the `step` value instead of the `37`. You also have a missing `;` here